frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/GitMacheteGraphTableManager.java

```diff
@@ -163,44 +172,52 @@ private Path getMacheteFilePath(GitRepository gitRepository) {
     return getGitDirectoryPath(gitRepository).resolve("machete");
   }
 
+  /**
+   * Repository update is queued as a background task, which in turn itself queues graph table refresh onto the UI thread.
+   */
   @Override
-  public void updateAndRefreshGraphTableInBackground() {
+  public void queueRepositoryUpdateAndGraphTableRefresh() {
```

Comment:
It's pretty important to introduce the distinction in the naming b/w the methods that **do** stuff, and the methods that **enqueue** stuff to be done (or enqueue conditionally - only if on the wrong thread; otherwise execute straight away).